tests/heka/test_message_parser.py

```diff
@@ -1,87 +1,104 @@
-                    count += 1
-            except Exception:
-                threw = True
-        assert expected_exceptions[t] == threw
 
+@pytest.mark.parametrize("heka_format", ["snappy", "gzip"])
+def test_telemetry(data_dir, heka_format):
+    filename = "{}/test_telemetry_{}.heka".format(data_dir, heka_format)
+    reference_filename = filename + '.json'
 
-top_keys = set(["application", "clientId", "creationDate", "environment", "id", "meta",
-                "payload", "type", "version"])
-payload_keys = set(["UIMeasurements", "addonDetails", "childPayloads", "chromeHangs",
-                    "fileIOReports", "histograms", "info", "keyedHistograms", "lateWrites",
-                    "log", "processes", "simpleMeasurements", "slowSQL", "threadHangStats",
-                    "ver", "webrtc"])
-
-
-def test_telemetry(data_dir):
-    filename = "{}/test_telemetry_gzip.heka".format(data_dir)
-    with open(filename, "rb") as o:
-        for r in message_parser.parse_heka_message(streaming_gzip_wrapper(o)):
-            assert set(r.keys()) == top_keys
-            assert set(r["payload"].keys()) == payload_keys
+    # enable this to regenerate the expected json representation of the ping
+    if False:
+        with open(filename, "rb") as f:
+            if "gzip" in heka_format:
+                f = streaming_gzip_wrapper(f)
+            # deep copy the parsed message so lazy-parsed json gets vivified
+            msg = copy.deepcopy(message_parser.parse_heka_message(f).next())
+            open(reference_filename, 'w').write(json.dumps(msg, indent=4,
+                                                           sort_keys=True))
 
-    filename = "{}/test_telemetry_snappy.heka".format(data_dir)
-    with open(filename, "rb") as o:
-        for r in message_parser.parse_heka_message(o):
-            assert set(r.keys()) == top_keys
-            assert set(r["payload"].keys()) == payload_keys
+    reference = json.load(open(reference_filename))
+    with open(filename, "rb") as f:
+        if "gzip" in heka_format:
+            f = streaming_gzip_wrapper(f)
+        # deep copy the parsed message so lazy-parsed json gets vivified
+        msg = copy.deepcopy(message_parser.parse_heka_message(f).next())
+        assert msg == reference
 
 
 def test_invalid_utf8(data_dir):
     filename = "{}/test_invalid_utf8.heka".format(data_dir)
     with open(filename, "rb") as o:
         for r in message_parser.parse_heka_message(o):
             assert(u'\ufffd' in r['info']['adapterDescription'])
+
+
+def test_lazy_parsing(data_dir, monkeypatch):
+    from moztelemetry.heka.message_parser import _parse_json
+    mock_parse_json = MagicMock(name='_parse_json',
+                                wraps=_parse_json)
+    import moztelemetry.heka.message_parser
+    monkeypatch.setattr(moztelemetry.heka.message_parser, '_parse_json', mock_parse_json)
+
+    # this heka message has 19 json fields, none of which should be parsed on
```

Comment:
Ah yes, you're right. Sorry, I misread the code. I'll update the comments appropriately.